engines/bops_api/app/jobs/bops_api/planning_application_dependency_job.rb

```diff
@@ -176,7 +200,11 @@ def fetch_entity(source)
 
       response.body
     rescue Faraday::ResourceNotFound
-      nil
+      "not_found"
+    rescue Faraday::ServerError, BopsApi::Errors::InvalidEntityResponseError
+      "failed"
+    rescue Faraday::ClientError => error
+      (error.response_status == 410) ? "removed" : "failed"
```

Comment:
Looking at the planning data server a 404 request returns a HTML page but does have the correct status. I think we should remove the `faraday.response :raise_error` line and check the status of the response. We should also rescue the generic `Faraday::Error` and re-raise with our own errors as returning these statuses means knowledge of how the constraint model works is leaking into our method to fetch entities. I suspect it would look something like this:

``` ruby
def fetch_entity(source)
  uri = normalise_url(source)

  return unless uri

  connection = Faraday.new(uri.origin) do |faraday|
    faraday.response :json, content_type: /\bjson$/
  end

  response = connection.get(uri.request_uri)
  
  case response.status
  when 200
    if response.body.is_a?(Hash)
      response.body
    else
      raise BopsApi::Errors::InvalidEntityResponseError, "Request for entity #{uri} returned a non-JSON response"
    end
  when 204
    nil
  when 404
    raise BopsApi::Errors::EntityNotFoundError, "Entity #{uri} was not found"
  when 410
    raise BopsApi::Errors::EntityRemovedError, "Entity #{uri} has been removed"
  else
    raise BopsApi::Errors::EntityFetchFailedError, "Entity #{uri} could not be fetched"
  end
rescue Faraday::Error
  raise BopsApi::Errors::EntityFetchFailedError, "Entity #{uri} could not be fetched"
end
```

You'd then update the `fetch_constraint_entities` method to update the status like this:

``` ruby
def fetch_constraint_entities(planning_application_constraint, entities)
  planning_application_constraint.update!(data: fetch_entities(entities), status: "success")
rescue BopsApi::Errors::EntityNotFoundError
  planning_application_constraint.update!(data: [], status: "not_found")
rescue BopsApi::Errors::EntityRemovedError
  planning_application_constraint.update!(data: [], status: "removed")
rescue BopsApi::Errors::EntityFetchFailedError
  planning_application_constraint.update!(data: [], status: "failed")
end
```

You'd also need to add the additional error classes and make `BopsApi::Errors::InvalidEntityResponseError` inherit from `BopsApi::Errors::EntityFetchFailedError` so the final `rescue` clause above captures the general case and the specific non-JSON response case.